tests/functional/olp-cpp-sdk-authentication/AuthenticationCommonTestFixture.cpp

```diff
@@ -0,0 +1,311 @@
+    const AuthenticationClient::SignInUserResponse& precond_failed_response,
+    bool do_cancel) {
+  AuthenticationCredentials credentials(id_, secret_);
+
+  std::shared_ptr<AuthenticationClient::SignInUserResponse> response;
+  unsigned int retry = 0u;
+  do {
+    if (retry > 0u) {
+      OLP_SDK_LOG_WARNING(__func__, "Request retry attempted (" << retry
+                                                                << ")");
+      std::this_thread::sleep_for(
+          std::chrono::seconds(retry * kRetryDelayInSecs));
+    }
+
+    std::promise<AuthenticationClient::SignInUserResponse> request;
+    auto request_future = request.get_future();
+    auto cancel_token = client_->AcceptTerms(
+        credentials,
+        precond_failed_response.GetResult().GetTermAcceptanceToken(),
+        [&request](const AuthenticationClient::SignInUserResponse& resp) {
+          request.set_value(resp);
+        });
+
+    if (do_cancel) {
+      cancel_token.cancel();
+    }
+
+    request_future.wait();
+    response = std::make_shared<AuthenticationClient::SignInUserResponse>(
+        request_future.get());
+  } while ((!response->IsSuccessful()) && (++retry < kMaxRetryCount) &&
+           !do_cancel);
+
+  return *response;
+}
+
+AuthenticationUtils::DeleteUserResponse
+AuthenticationCommonTestFixture::DeleteUser(
+    const std::string& user_bearer_token) {
+  std::shared_ptr<AuthenticationUtils::DeleteUserResponse> response;
+  unsigned int retry = 0u;
+  do {
+    if (retry > 0u) {
+      OLP_SDK_LOG_WARNING(__func__, "Request retry attempted (" << retry
+                                                                << ")");
+      std::this_thread::sleep_for(
+          std::chrono::seconds(retry * kRetryDelayInSecs));
+    }
+
+    std::promise<AuthenticationUtils::DeleteUserResponse> request;
+    auto request_future = request.get_future();
+    utils_->DeleteHereUser(
+        *network_, olp::http::NetworkSettings(), user_bearer_token,
+        [&request](const AuthenticationUtils::DeleteUserResponse& resp) {
+          request.set_value(resp);
+        });
+    request_future.wait();
+    response = std::make_shared<AuthenticationUtils::DeleteUserResponse>(
+        request_future.get());
+  } while ((response->status < 0) && (++retry < kMaxRetryCount));
+
+  return *response;
+}
+
+AuthenticationClient::SignOutUserResponse
+AuthenticationCommonTestFixture::SignOutUser(const std::string& access_token,
+                                             bool do_cancel) {
+  AuthenticationCredentials credentials(id_, secret_);
+  std::promise<AuthenticationClient::SignOutUserResponse> request;
+  auto request_future = request.get_future();
+  auto cancel_token = client_->SignOut(
+      credentials, access_token,
+      [&](const AuthenticationClient::SignOutUserResponse& response) {
+        request.set_value(response);
+      });
+
+  if (do_cancel) {
+    cancel_token.cancel();
+  }
+
+  request_future.wait();
```

Comment:
wait_for or just plain get maybe?

Reply:
Looks like this wait is not needed at all - request_future.get() is sufficient here. I will add this in a separate PR.